src/brucke_health_check.erl

```diff
@@ -0,0 +1,131 @@
+-define(DOWNSTREAM, <<"downstream">>).
+-define(PRODUCERS, <<"producers">>).
+-define(CONSUMERS, <<"consumers">>).
+-define(STATUS, <<"status">>).
+-define(MEMBERS, <<"members">>).
+-define(TOPIC, <<"topic">>).
+-define(ID, <<"id">>).
+-define(PARTITION, <<"partition">>).
+
+-include("brucke_int.hrl").
+
+%% API
+-export([init/0, get_report_data/1]).
+
+init() ->
+  case application:get_env(?APPLICATION, healthcheck) of
+    {ok, true} ->
+      Port = application:get_env(?APPLICATION, healthcheck_port, 8080),
+      Dispatch = cowboy_router:compile([{'_', [{'_', brucke_http_handler, []}]}]),
+      {ok, _} = cowboy:start_clear(http, 1, [{port, Port}], #{env => #{dispatch => Dispatch}}),
+      ok;
+    _ -> ok
+  end.
+
+-spec get_report_data(proplists:proplist()) -> map().
+get_report_data(Params) ->
+  AskClients = proplists:get_value(<<"clients">>, Params, true),
+  AllCliends = get_all_clients(AskClients),
+  AskRoutes = proplists:get_value(<<"routes">>, Params, true),
+  AllRoutes = get_all_routes(AskRoutes),
+  #{<<"clients">> => AllCliends, <<"routes">> => AllRoutes}.
+
+
+%% @private
+-spec get_all_clients(boolean()) -> list(map()).
+get_all_clients(true) ->
+  AllClients = brucke_config:all_clients(),
+  lists:map(
+    fun({ClientId, _, _}) ->
+      case whereis(ClientId) of
+        undefined -> #{ClientId => undefined};
+        _ -> #{ClientId => ok}
+      end
+    end, AllClients);
+get_all_clients(_) -> [].
+
+%% @private
+-spec get_all_routes(binary()) -> list(map()).
+get_all_routes(true) ->
+  AllRoutes = brucke_config:all_routes(),
+  Children = brucke_sup:get_children(),
+  lists:map(
+    fun(#route{upstream = Upstream, downstream = Downstream}) ->
+      Route = check_route(Upstream, Children),
+      UpMap = get_brod_data(Upstream, fun brod:get_consumer/3),
+      DownMap = get_brod_data(Downstream, fun brod:get_producer/3),
+      {UpstreamId, UpstreamTopic} = Upstream,
+      URoute = Route#{?UPSTREAM => #{?ID => UpstreamId, ?TOPIC => UpstreamTopic, ?CONSUMERS => UpMap}},
+      {DownstreamId, DownstreamTopic} = Downstream,
+      URoute#{?DOWNSTREAM => #{?ID => DownstreamId, ?TOPIC => DownstreamTopic, ?PRODUCERS => DownMap}}
+    end, AllRoutes);
+get_all_routes(_) -> [].
+
+%% @private
+check_route(Upstream, Routers) ->
+  case lists:keyfind(Upstream, 1, Routers) of
+    {_, Child, _, _} when is_pid(Child) ->  % ok / dead (but pid found in sup tree)
+      check_process(Child);
+    {_, Status, _, _} ->  % undefined / restarting
+      #{?STATUS => Status};
+    false ->  % not in tree
+      #{?STATUS => <<"not under brucke_sup">>}
+  end.
+
+%% @private
+check_process(Route) ->
+  case is_process_alive(Route) of
+    true ->
+      Members = get_all_brucke_members(Route),
+      #{?STATUS => ok, ?MEMBERS => Members};
+    false ->
+      #{?STATUS => <<"dead">>, ?MEMBERS => []}
+  end.
+
+%% @private
+get_brod_data({Id, Topic}, GetFun) ->
+  case brod:get_partitions_count(Id, Topic) of
```

Comment:
In case you wonder: `brucke_member` is always started and stands by even if no assignment is received.